app/models/Penguin.rb

```diff
@@ -0,0 +1,19 @@
+class Penguin < ActiveRecord::Base
+
+  def rookery_access
+    x = self.rookery_id
+    y = Rookery.find_by_id(x)
+    y
+  end
+
+  def event_access
+    x = self.event_id
+    y = Event.find_by_id(x)
+    y
+  end
+
+  def delete_results
+    Result.where("penguin_id" => self.id).first.delete
```

Comment:
`Result.where("penguin_id" => self.id)` returns a collection of all of a penguin's results. It would make sense that you get that collection in order to delete a penguin's results. So :+1: so far.

**However,** your code then continues to grab only the _first_ result from that collection:

```ruby
Result.where("penguin_id" => self.id).first
```

and it deletes that.

But the goal is to delete all of those results.

This code should be either:

```ruby
Result.where("penguin_id" => self.id).each do |single_result|
  single_result.delete
end
```

or

```ruby
Result.where("penguin_id" => self.id).delete_all
```

Both accomplish the same goal.